libclamav/htmlnorm.c

```diff
@@ -1189,16 +1189,21 @@ static bool cli_html_normalise(cli_ctx *ctx, int fd, m_area_t *m_area, const cha
                             /*don't output newlines in nocomment.html
                              * html_output_c(file_buff_o2, '\n');*/
                         } else if ((strcmp(tag, "/style") == 0) && (in_tag == TAG_STYLE)) {
```

Comment:
"</style>" could be defined as a constant in this file, which would also save you having to run `strlen()` on it constantly.  This is a technique I've done for these sorts of "hand-coded" parsers before:
``` c
#include <string.h>
#include <assert.h>

static const char STYLE_TAG[] = "</style>";
static const unsigned long STYLE_TAG_LEN = sizeof(STYLE_TAG) - 1;

int main()
{
    assert(strlen(STYLE_TAG) == STYLE_TAG_LEN);
}
```

Reply:
The compiler will optimize out `sizeof` and `strlen`'s of static strings, meaning this change
- won't have a performance benefit,
- adds lines of code to the file, 
- and requires anyone who wants to know what it's searching for to cross-reference the constant. 

Unless I'm missing a benefit to it, I'm not a big fan of the change.

Reply:
Ah, I actually wasn’t aware that Clang (and maybe others) specifically optimize out calls to `strlen()`.  I just see those and think, “library call — optimization dead-end”.  Thanks for clueing me on that.

`sizeof()` is actually a C operator.

Parser authors commonly define constants for all the English-readable stuff you might find in an encoded format like HTML (right down to things like `LEFT_ANGLE_BRACKET`), which is where I probably borrowed that technique from.